src/templates/home/components/queryForm/QueryForm.js

```diff
@@ -3,16 +3,18 @@ import { Row, Col } from 'react-bootstrap';
 
 
 const QueryForm = (props) => {
-    const { formIndication, interactionTypes} = props;
+    const { formIndication, interactionTypes } = props;
 
     /* Function for rendering the Interaction Types as select options */
-    function RenderInteractionOptions() {
+    function RenderInteractionOptions(setInteractionType) {
```

Comment:
Is this called every time the home page is rendered? Can you cache the result?

Reply:
In principle React caches these things itself if the state of the document is not changed, but otherwise it should rererender the content this way cause we made the interaction types dynamic